src/components/PhoneInput/PhoneInput.jsx

```diff
@@ -0,0 +1,131 @@
+// @flow
+import React, {
+  Component,
+  type Node,
+  type ComponentType,
+} from 'react';
+import { PhoneInput as ReactPhoneInput } from 'react-phone-number-input';
+import { SmartInput } from 'react-phone-number-input/smart-input';
+import { getExampleNumber } from 'libphonenumber-js';
+
+import metadata from 'libphonenumber-js/metadata.min.json';
+import examples from 'libphonenumber-js/examples.mobile.json';
+import labels from 'react-phone-number-input/locale/default.json';
+
+import { getDataAttrs } from '../../dataUtils';
+import type { Data } from '../../types';
+
+import CountrySelectComponent from './CountrySelectComponent';
+import FlagComponent from './FlagComponent';
+
+import type {
+  Metadata,
+  PhoneNumber,
+} from './types';
+
+import styles from './style.scss';
+
+export type InputComponentProps = {|
+  value: string,
+  onChange?: (value: ?string) => void,
+  onFocus?: () => void,
+  onBlur?: () => void,
+  country?: ?string,
+  metadata: Metadata,
+|};
+
+type State = {|
+  country: ?string,
+|};
+
+export type Props = {|
+  value?: ?string,
+  onChange?: (value: ?string) => void,
+  onFocus?: <+T: EventTarget = EventTarget>(SyntheticFocusEvent<T>) => void,
+  onBlur?: <+T: EventTarget = EventTarget>(SyntheticFocusEvent<T>) => void,
+  onKeyDown?: <+T: EventTarget = EventTarget>(SyntheticKeyboardEvent<T>) => void,
+  disabled: boolean,
+  autoComplete: string,
+  displayInitialValueAsLocalNumber: boolean,
+  country?: string,
+  countries?: Array<string>,
+  labels: Labels,
+  flagsPath: string,
+  flags: Flags,
+  flagComponent: ComponentType<$Shape<FlagComponentProps>>,
+  international: boolean,
+  internationalIcon: () => ComponentType<*>,
+  showCountrySelect: boolean,
+  countrySelectTabIndex?: number,
+  countryOptions?: Array<string>,
+  style?: { [key: string]: string },
+  className?: string,
+  inputClassName?: string,
+  getInputClassName?: ({| disabled?: boolean, invalid?: boolean |}) => string,
+  countrySelectComponent: ComponentType<$Shape<CountrySelectComponentProps>>,
+  inputComponent: ComponentType<$Shape<InputComponentProps>>,
+  ext?: Node,
+  limitMaxLength: boolean,
+  error?: string,
+  indicateInvalid?: boolean,
+  locale?: { [key: string]: string },
+  metadata: Metadata,
+  onCountryChange: (country: ?string) => void,
+  placeholder?: string,
+  data: Data,
+  [prop: string]: any,
+|};
+
+function generateExampleNumber(country: ?string): string {
```

Comment:
I generally try to avoid adding too many functions into components for a few reasons.

For example, all functions on a JS class are (currently) public, which I prefer this one to not be.

And another reason is that this function is pure (it doesn't require the use of `this`), and therefore can be static, which means it doesn't really have it to be part of the component (especially if I want to keep it private).